File: server/models/survey.js
const mongoose = require('mongoose')
const surveySchema = new mongoose.Schema({
    title:String,
    subtitle:String,
    imageSubtitle:String,
    question: Array,
    owner: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User'
    },
    responses:Array,
    usedIps: Array,
    date: Date,
    public: Boolean,
    
  })
  surveySchema.set('toJSON', {
    transform: (document, returnedObject) => {
      returnedObject.id = returnedObject._id.toString()
      delete returnedObject._id
      delete returnedObject.__v
    }
  })
  
  const Survey = mongoose.model('Survey', surveySchema)
  module.exports = Survey

File: dotnet-server/dotnet-server/Models/Survey.cs
using System.ComponentModel.DataAnnotations;

namespace dotnet_server.Models;

public class Survey
{
    [Key]
    public string Id { get; set; }
    public string Title { get; set; }
    public ICollection<Question> Questions { get; set; }
    public User Owner { get; set; }
    public ICollection<SurveyResponse> SurveyResponses { get; set; }

    public ICollection<UsedIps> UsedIps { get; set; }

    public DateTime Date { get; set; }

    public bool IsPublic { get; set; }
}
